### PR TITLE
feat(smpr): Config struct + TOML/.env loading (#67, #68)

### LINKED ISSUE #67: Implement Config struct and TOML/.env loading
Port `build_config`, `load_toml_config`, `load_env` to Rust. Implement the Config struct with serde deserialization from TOML. CLI flags > env > .env > TOML > defaults precedence. Include `ServerConfig`, forced_libraries, detection word lists.

### LINKED ISSUE #68: Implement CLI argument parsing with clap derive
Port `build_parser` to clap derive macros. Subcommands: rate, force, reset, configure. Shared options via clap's arg groups. Match the existing Python CLI interface exactly.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Implement `Config` struct with TOML/.env loading and CLI arg merging for the Rust rewrite (`smpr/`)
- Two-type architecture: `RawConfig` (serde deserialization, all `Option` fields) and `Config` (fully resolved, validated)
- Merge precedence: CLI flags > env vars > `.env` file > TOML > hardcoded defaults
- Per-server library/location `force_rating` support via nested `BTreeMap`s
- `CliInput` bridge struct decouples config module from clap
- 19 unit tests covering happy path, error cases, and edge cases
- UAT verified against Jellyfin at localhost:8097

## Details

- Bumps MSRV to 1.94
- `ConfigError` enum with `Display`/`Error` impls for all failure modes
- Detection word lists resolve independently (partial TOML overrides work)
- One-off server mode (`--server-url` + `--api-key`) bypasses TOML servers
- Explicit `--config`/`--env-file` paths error if missing; default paths warn/skip

Closes #67, closes #68

## Test plan

- [x] `cargo test -- config::tests --test-threads=1` — 19/19 passing
- [x] `cargo clippy -- -D warnings` — clean
- [x] `cargo fmt -- --check` — clean
- [x] UAT: `smpr rate --config explicit_config.uat.toml --env-file .env --server uat-jellyfin -v --dry-run`
- [x] UAT: `smpr rate --server-url http://localhost:8097 --api-key <key> -v --dry-run`
- [x] UAT: error cases (wrong server name, no config, missing env file)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Comprehensive configuration system supporting TOML, .env, and CLI inputs with server/library selection, detection settings, and resolved precedence
  * Built-in moderation word lists (offensive and PG-13 categories plus curated false-positive entries)
  * Enhanced rate/force/reset commands to load and display resolved configuration

* **Tests**
  * Extensive configuration parsing and loading tests covering many edge cases

* **Chores**
  * Rust toolchain bumped to 1.94; added a dev test dependency and constrained CI tests to a single thread

### CODE REVIEW

File: smpr/src/config/tests.rs [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+[servers.nokey-server]
+url = "http://localhost:8096"
+"#;
+    let mut toml_file = NamedTempFile::new().unwrap();
+    toml_file.write_all(toml_content.as_bytes()).unwrap();
+
+    // Ensure the env var does NOT exist
+    unsafe {
+        std::env::remove_var("NOKEY_SERVER_API_KEY");
+    }
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply smpr/src/config/tests.rs]: Fixed in 7284893. CI now passes `--test-threads=1` to `cargo test` so env-var-mutating tests run serially.

File: smpr/src/config/mod.rs [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+                    "unknown server '{requested}' in --server filter. Available: {}",
+                    available.join(", ")
+                )
+            }
+            Self::NoServers => write!(f, "no servers configured"),
+        }
+    }
+}
+
+impl std::error::Error for ConfigError {}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply smpr/src/config/mod.rs]: Tracked as #99 (Rust Rewrite: Foundation milestone). Will implement `source()` when we add structured error reporting.
